capa/features/extractors/dnfile/insn.py

```diff
@@ -57,68 +62,182 @@ def get_methods(ctx: Dict) -> Dict:
     return ctx["methods_cache"]
 
 
-def get_callee(ctx: Dict, token: int) -> Union[DnMethod, DnUnmanagedMethod, None]:
+def get_callee(ctx: Dict, token: int) -> Union[DnType, DnUnmanagedMethod, None]:
     """map dotnet token to un/managed method"""
-    callee: Union[DnMethod, DnUnmanagedMethod, None] = get_managed_imports(ctx).get(token, None)
-    if not callee:
+    callee: Union[DnType, DnUnmanagedMethod, None] = get_managed_imports(ctx).get(token, None)
+    if callee is None:
         # we must check unmanaged imports before managed methods because we map forwarded managed methods
         # to their unmanaged imports; we prefer a forwarded managed method be mapped to its unmanaged import for analysis
         callee = get_unmanaged_imports(ctx).get(token, None)
-        if not callee:
+        if callee is None:
             callee = get_methods(ctx).get(token, None)
     return callee
 
 
+def get_properties(ctx: Dict) -> Dict:
+    if "properties_cache" not in ctx:
+        ctx["properties_cache"] = {}
+        for prop in get_dotnet_properties(ctx["pe"]):
+            ctx["properties_cache"][prop.token] = prop
+    return ctx["properties_cache"]
+
+
+def get_fields(ctx: Dict) -> Dict:
+    if "fields_cache" not in ctx:
+        ctx["fields_cache"] = {}
+        for field in get_dotnet_fields(ctx["pe"]):
+            ctx["fields_cache"][field.token] = field
+    return ctx["fields_cache"]
+
+
 def extract_insn_api_features(fh: FunctionHandle, bh, ih: InsnHandle) -> Iterator[Tuple[Feature, Address]]:
     """parse instruction API features"""
     insn: Instruction = ih.inner
 
     if insn.opcode not in (OpCodes.Call, OpCodes.Callvirt, OpCodes.Jmp, OpCodes.Calli):
         return
 
-    callee: Union[DnMethod, DnUnmanagedMethod, None] = get_callee(fh.ctx, insn.operand.value)
+    callee: Union[DnType, DnUnmanagedMethod, None] = get_callee(fh.ctx, insn.operand.value)
     if callee is None:
         return
 
-    if isinstance(callee, DnUnmanagedMethod):
-        # like kernel32.CreateFileA
-        for name in capa.features.extractors.helpers.generate_symbols(callee.modulename, callee.methodname):
-            yield API(name), ih.address
-    else:
+    if isinstance(callee, DnType):
+        if callee.member.startswith(("get_", "set_")):
+            if insn.operand.table == METHODDEF_TABLE:
+                """check if the method belongs to the MethodDef table and whether it is used to access a property"""
```

Comment:
```suggestion
                # check if the method belongs to the MethodDef table and whether it is used to access a property
```

Reply:
this worked fine as-is, but is inconsistent with the rest of the code base today